drivers/clock_control/nrf_power_clock.c

```diff
@@ -9,272 +9,311 @@
 #include <errno.h>
 #include <sys/atomic.h>
 #include <device.h>
-#include <drivers/clock_control.h>
 #include <sys/__assert.h>
+#include <drivers/clock_control.h>
+#include <drivers/clock_control/nrf_clock_control.h>
 #include <nrf_clock.h>
-#if defined(CONFIG_USB) && defined(CONFIG_SOC_NRF52840)
+#include <logging/log.h>
+#if defined(CONFIG_USB_NRF52840) || NRF_POWER_HAS_POFCON
 #include <nrf_power.h>
-#include <drivers/clock_control/nrf_clock_control.h>
 #endif
 
-static u8_t m16src_ref;
-static u8_t m16src_grd;
-static u8_t k32src_initialized;
+LOG_MODULE_REGISTER(clock_control, CONFIG_CLOCK_CONTROL_LOG_LEVEL);
+
+/* Helper logging macros which prepends device name to the log. */
+#define CLOCK_LOG(lvl, dev, ...) \
+	LOG_##lvl("%s: " GET_ARG1(__VA_ARGS__), dev->config->name \
+			COND_CODE_0(NUM_VA_ARGS_LESS_1(__VA_ARGS__),\
+					(), (, GET_ARGS_LESS_1(__VA_ARGS__))))
+#define ERR(dev, ...) CLOCK_LOG(ERR, dev, __VA_ARGS__)
+#define WRN(dev, ...) CLOCK_LOG(WRN, dev, __VA_ARGS__)
+#define INF(dev, ...) CLOCK_LOG(INF, dev, __VA_ARGS__)
+#define DBG(dev, ...) CLOCK_LOG(DBG, dev, __VA_ARGS__)
+
+/* returns true if clock stopping must be deferred due to active calibration. */
+typedef void (*nrf_clock_stop_handler_t)(struct device *dev);
+
+/* Clock instance structure */
+struct nrf_clock_control {
+	atomic_t ref;		/* Users counter */
```

Comment:
Why is it defined as `atomic_t` when atomic API is not used for this field?

Reply:
changed to `s32_t`